packages/client/modules/newTeam/components/NewTeamForm/NewTeamForm.tsx

```diff
@@ -146,96 +124,88 @@ class NewTeamForm extends Component<Props, State> {
     return teamNameValidation(rawTeamName, teamNames)
   }
 
-  validateOrgName = () => {
-    const {fields} = this.state
-    const rawOrgName = fields.orgName.value
-    return new Legitity(rawOrgName)
-      .trim()
-      .required('Your new org needs a name!')
-      .min(2, 'C’mon, you call that an organization?')
-      .max(100, 'Maybe just the legal name?')
+  const validate = (name: FieldName) => {
+    const validators = {
+      teamName: validateTeamName,
+      orgName: validateOrgName
+    }
+    const res: Legitity = validators[name]()
+    const field = fields[name]
+    if (res.error !== field.error) {
+      setFields({
```

Comment:
nit: we've got a nice hook called `useForm` that takes care of all this boilerplate logic for you if you'd like to use that